src/engraving/dom/note.cpp

```diff
@@ -1405,6 +1405,50 @@ void Note::updateFrettingForTiesAndBends()
     setFret(prevNote->fret());
 }
 
+bool Note::hideFret() const
+{
+    if (!tieBack() || forceShowFret()) {
+        return false;
+    }
+
+    ShowTiedFret showTiedFret = style().value(Sid::tabShowTiedFret).value<ShowTiedFret>();
+    if (showTiedFret == ShowTiedFret::TIE_AND_FRET) {
```

Comment:
Actually to be very honest, I'm not sure if these methods are as clear as they could be. A few things that may be the cause of this are:
- the fact that `Sid::tabShowTiedFret` is queried in a lot of places (both in forceShowFret itself, _and_ by the callers of forceShowFret) makes the logic a bit less easy to grasp. 
- Patterns like the following seem to be not my favourite:
    ```
    if (it is A) 
        return …
    if (it is C) 
        return …

    // then it must be B
    return …
    ```
    A simple switch might be slightly nicer:
    ```
    case A: return …
    case B: return …
    case C: return …
    ```
- It's not immediately clear which parameter takes precedence over the others, so to speak. This might become clearer by using switches. Maybe there _is_ simply some interaction between the parameters instead of a strict precedence hierarchy, which is fine; but I think it would be possible to write it down slightly clearer.

Anyway, not sure if this is worth spending time on right now, but in the refinement stage you could keep this in mind.

Reply:
These methods look cumbersome because they implement cumbersome logic (which I deeply hate, but I also can't do much about). I honestly couldn't find a clearer way to do this. The logic isn't simple enough that a switch statement would make things better imo, and in these cases with intricate relations I do prefer the "early return" pattern (the alternative is a nesting chain of hell).